__tests__/server/plugins/ensureCorrelationId.spec.js

```diff
@@ -14,54 +14,52 @@
  * permissions and limitations under the License.
  */
 
-const ensureCorrelationId = require('../../../src/server/middleware/ensureCorrelationId').default;
+import ensureCorrelationId from '../../../src/server/plugins/ensureCorrelationId';
 
 describe('ensureCorrelationId', () => {
   function generateSamples() {
-    const req = {
+    const request = {
       headers: {},
     };
-    const res = null;
-    const next = jest.fn();
-    return { req, res, next };
+
+    const fastify = {
+      addHook: jest.fn((_hookName, cb) => {
+        cb(request);
+      }),
+    };
+
+    return { fastify, request };
   }
 
   it('uses correlation_id for correlation-id on a request without correlation-id', () => {
-    const { req, res, next } = generateSamples();
-    delete req.headers['correlation-id'];
-    req.headers.correlation_id = 'cloudy';
-    ensureCorrelationId(req, res, next);
-    expect(req.headers).toHaveProperty('correlation-id', 'cloudy');
+    const { fastify, request } = generateSamples();
+    delete request.headers['correlation-id'];
+    request.headers.correlation_id = 'cloudy';
+    ensureCorrelationId(fastify, null, jest.fn());
+    expect(request.headers).toHaveProperty('correlation-id', 'cloudy');
   });
 
   it('uses unique_id for correlation-id on a request without correlation-id or correlation_id', () => {
-    const { req, res, next } = generateSamples();
-    delete req.headers['correlation-id'];
-    delete req.headers.correlation_id;
-    req.headers.unique_id = 'thunderstorms';
-    ensureCorrelationId(req, res, next);
-    expect(req.headers).toHaveProperty('correlation-id', 'thunderstorms');
+    const { fastify, request } = generateSamples();
+    delete request.headers['correlation-id'];
+    delete request.headers.correlation_id;
+    request.headers.unique_id = 'thunderstorms';
+    ensureCorrelationId(fastify, null, jest.fn());
+    expect(request.headers).toHaveProperty('correlation-id', 'thunderstorms');
   });
 
   it('adds a correlation-id header to a request without anything', () => {
```

Comment:
```suggestion
  it('adds a unique correlation-id to request header if correlation-id is null', () => {
```

Reply:
I propose `adds a unique correlation-id to the request object` since tests are assertions, things like `without` and `if` shouldn't be used. Also, these are the original test cases, maybe we should spend some time refreshing them 🤔